### ISSUE TITLE
cut/copy & paste not working even i checked 'allow' for clipbored permission 

### ISSUE THREAD

**Issue by arshadirfan** [photopea/photopea#2104]
cut/copy& paste not working with any usual method. first time when pressed ctrl+c then ctrl+v, the page asked permission to access the clipboard. I checked 'allow'. after that nothing happens when I try to copy-paste by both usual methods. help me to resolve the problem

**Comment by photopea** [comment]
Hi, can you disable all browser extensions and try it again? Are you using Photopea at www.Photopea.com ?

**Comment by photopea** [comment]
I have to close it because of lack of details.

**Comment by arshadirfan** [comment]
> Hi, can you disable all browser extensions and try it again? Are you using Photopea at [www.Photopea.com](http://www.Photopea.com) ?

sorry, i miised your replay.
I tried www.photopea.com by disabling all extention in chrome,but  i got no result.
i tried photopea in installed app with chrome,  microsoft edge in my pc, and also google chrome in my android phone, same problom exists everywhere.

**Comment by photopea** [comment]
Can you update your web browser? Can you check that your permissions are allowed? Do you copy paste with Ctrl+C and Ctlr+V ?

**Comment by arshadirfan** [comment]
I am using latest version of chrome.
I tried both  Ctrl+C and Ctlr+V method and edit>copy and edit>paste method. none of them are working

**Comment by photopea** [comment]
Can you try if it works in Incognito mode? Also, do you have a selection when copying?

**Comment by arshadirfan** [comment]
I tried incognito mode, not working. Also I had selection of layer when copying. I tried it in another pc, same problom exists.

are you able to copy in your device?

**Comment by photopea** [comment]
Yes, I am able to copy and paste. Note, that when you copy, nothing will happen. Only after you paste, the new layer will be added (but the image will look the same).

**Comment by arshadirfan** [comment]
I know that. i still uses same methode in photoshop CC.
Now I found that this problom happen only when direct copying of a layer.
it works when copying selceted a part of layer with rectangular selction tool and that become succefully pasted as new layer. also no problom with text copying . Note that its  not working for 'text layer'.

**Comment by photopea** [comment]
To copy a part of the layer, there needs to be a selection. Without it, nothing is copied. This works the same in Photoshop and Photopea.

Could you be more specific, what exactly should I do, what works different in Photopea than in Photoshop?

**Comment by arshadirfan** [comment]
you know auto-selection enabling option with move tool. with that I selected a layer by simply clicking on it. in the layer panel, it is clearly seen that the layer is selected.
     now  in photoshop, a duplicate later appears by doing  copy and paste
     but in photopea , it is not appearing a duplicate of that layer by doing copy and paste commands.

**Comment by photopea** [comment]
Oh, I did not know about this feature of Photoshop (copying and pasting duplicates layers, when there is no selection). 

The problem is, that e.g. if you do a PrintScreen, and also click on a layer in Photopea and press Ctrl+C, and then you want to paste, it is not clear, which one should be pasted (a screenshot or a duplicate of a layer). Pasting both would be confusing. The one that was copied later should be pasted, but the browser does not let us know, at what time the screenshot was taken.

I asked Google Developers, if it is possible to make Photopea clear the image clipboard of the computer (so at Ctrl+C in Photopea, we delete any possible screenshot in a clipboard, so that it can not be pasted).

**Comment by arshadirfan** [comment]
OK. I think the missing feature is one of the important things for every graphic editing platform.  Best wishes for your attempt to add that feature. 
thank you.

**Comment by photopea** [comment]
You can press Ctrl+J to duplicate layers, it will do the same thing. I think that is what most of people use in Photoshop.

**Comment by arshadirfan** [comment]
ok, it is a new thing for me, Thank you
